run-build-functions.sh

```diff
@@ -645,6 +653,22 @@ install_dependencies() {
     fi
   fi
 
+  # Install rust dependencies and compile the code.
+  if [ -f Cargo.toml ] || [ -f Cargo.lock ]
+  then
+    restore_home_cache ".cargo/registry" "rust deps"
+    check_rust_target_cacheability && restore_home_cache "repo/target" "rust compile output"
+    source $HOME/.cargo/env
+    cargo build --release
```

Comment:
Totally agree! Should I remove the entire `if` block?

Reply:
i think it makes sense to keep the cache handling.
if `cargo` had a command to download dependencies, but not build the project, we could execute that here, since for most languages we seperate the install and build steps.

Reply:
I'm removing this so it will only restore the cache. Fixed in 60fd309

We don't pull the dependencies in Rust as they depend on the build flags.

I don't know what would be better if the dependencies were pulled separately.... let me know if something else should be done here.